src/mesh.cpp

```diff
@@ -3533,7 +3570,7 @@ int LibMesh::get_bin(Position r) const
 int LibMesh::get_bin_from_element(const libMesh::Elem* elem) const
 {
   int bin =
-    adaptive_ ? elem_to_bin_map_[elem->id()] : elem->id() - first_element_id_;
+    adaptive_ ? elem_to_bin_map_[clustering_element_mapping_.at(elem)->id()] : elem->id() - first_element_id_;
```

Comment:
`cluster_element_mapping` is only populated when `amalgamation_` is true, so shouldn't you always check whether `amalgamation_` is true?